lib/rebay_api_web/controllers/session_controller.ex

```diff
@@ -0,0 +1,50 @@
+defmodule RebayApiWeb.SessionController do
+  use RebayApiWeb, :controller
+  plug Ueberauth
+
+  alias RebayApi.Repo
+  alias RebayApi.Accounts.User
+
+  def create(%{assigns: %{ueberauth_auth: auth}} = conn, %{"provider" => provider}) do
+    user_params = user_params(auth, provider)
+    changeset = User.changeset(%User{}, user_params)
+
+    case insert_or_update_user(changeset) do
+      {:ok, user} ->
+        conn
+        |> put_session(:id, auth.credentials.token)
+        |> put_session(:user_uuid, user.uuid)
+        |> put_resp_cookie("session_id", auth.credentials.token, [http_only: true])
+        |> redirect(external: "#{System.get_env("CLIENT_HOST")}/login/#{user.uuid}")
```

Comment:
I'd recommend pushing this config into something checked at startup, otherwise you run the risk of only failing during request/response flow.

Reply:
I don't think I understand what you mean by this. If a user doesn't have to be authenticated at startup, then how does doing anything wrt authentication during startup help?

Reply:
Yeah, I could have been more clear. I was talking about the `System.get_env` call. Your system should fail startup if that env variable doesn't exist, otherwise you will only see errors about it during runtime.

Reply:
Got it, thanks for clarifying. Addressed in PR #10